sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -55,6 +82,14 @@ private[joins] final class UniqueKeyHashedRelation(hashTable: JavaHashMap[Row, R
   }
 
   def getValue(key: Row): Row = hashTable.get(key)
+
+  override def writeExternal(out: ObjectOutput): Unit = {
+    writeBytes(out, SparkSqlSerializer.serialize(hashTable))
```

Comment:
This is probably very huge for `SparkSqlSerializer.serialize(hashTable)`, as in `ShuffledHashJoin`, we will load the whole partition data into the `hashTable`, which mean we probably keep 2 copies of the data in the memory, will that the intention?